Library/Homebrew/brew.rb

```diff
@@ -158,10 +168,13 @@ class MissingEnvironmentVariables < RuntimeError; end
   exit 1
 rescue Exception => e # rubocop:disable Lint/RescueException
   onoe e
-  if internal_cmd && defined?(OS::ISSUES_URL) &&
-     !ENV["HOMEBREW_NO_AUTO_UPDATE"]
-    $stderr.puts "#{Tty.bold}Please report this issue:#{Tty.reset}"
-    $stderr.puts "  #{Formatter.url(OS::ISSUES_URL)}"
+  if internal_cmd && !ENV["HOMEBREW_NO_AUTO_UPDATE"]
+    if defined?(OS::ISSUES_URL)
+      $stderr.puts "#{Tty.bold}Please report this issue:#{Tty.reset}"
+      $stderr.puts "  #{Formatter.url(OS::ISSUES_URL)}"
+    elsif Homebrew.args.HEAD?
+      head_unsupported_error
+    end
```

Comment:
```suggestion
  if internal_cmd && !ENV["HOMEBREW_NO_AUTO_UPDATE"]
    if Homebrew.args.HEAD?
      head_unsupported_error
    elsif defined?(OS::ISSUES_URL)
      $stderr.puts "#{Tty.bold}Please report this issue:#{Tty.reset}"
      $stderr.puts "  #{Formatter.url(OS::ISSUES_URL)}"
    end
```

Reply:
I'm not sure this case is needed? Is `BuildError` above insufficient?

Reply:
I couldn't find and test every possible permutation of `brew install --HEAD` errors to find out what exceptions they raise (it was enough playing whackamole with the patches and "normal" ones). If anything trickles down to the base `Exception` then it'll get this message (ie if it's not a patch, or a normal build error). If this isn't needed then I can get rid of it.

Reply:
How about we omit this one for now and see if we get any reports?

Reply:
Sure.

Reply:
Done!